# Finding out all the genres types
This script was used to determine all the different genres contained in our genres string. Obviously with so many rows of data and multiple genres for a lot of the movies I felt the best way to go about this was to look at the data and iterate through the genre column looking for unique values

In [1]:
import pandas as pd

In [2]:
movies = pd.read_csv('movies.csv')
movies

movieId                                      title  \
0           1                           Toy Story (1995)   
1           2                             Jumanji (1995)   
2           3                    Grumpier Old Men (1995)   
3           4                   Waiting to Exhale (1995)   
4           5         Father of the Bride Part II (1995)   
...       ...                                        ...   
9737   193581  Black Butler: Book of the Atlantic (2017)   
9738   193583               No Game No Life: Zero (2017)   
9739   193585                               Flint (2017)   
9740   193587        Bungo Stray Dogs: Dead Apple (2018)   
9741   193609        Andrew Dice Clay: Dice Rules (1991)   

                                           genres  
0     Adventure|Animation|Children|Comedy|Fantasy  
1                      Adventure|Children|Fantasy  
2                                  Comedy|Romance  
3                            Comedy|Drama|Romance  
4                                          Comedy  
...                                           ...  
9737              Action|Animation|Comedy|Fantasy  
9738                     Animation|Comedy|Fantasy  
9739                                        Drama  
9740                             Action|Animation  
9741                                       Comedy  

[9742 rows x 3 columns]

In [3]:
movies['genres'].value_counts()

Drama                                          1053
Comedy                                          946
Comedy|Drama                                    435
Comedy|Romance                                  363
Drama|Romance                                   349
                                               ... 
Adventure|Drama|Thriller|IMAX                     1
Action|Crime|Sci-Fi|Thriller                      1
Comedy|Fantasy|Romance|Sci-Fi                     1
Action|Adventure|Drama|Fantasy|Mystery|IMAX       1
Action|Adventure|Crime|Drama                      1
Name: genres, Length: 951, dtype: int64

In [4]:
genres = movies['genres'].str.split('|', expand=False)
genres

0       [Adventure, Animation, Children, Comedy, Fantasy]
1                          [Adventure, Children, Fantasy]
2                                       [Comedy, Romance]
3                                [Comedy, Drama, Romance]
4                                                [Comedy]
                              ...                        
9737                 [Action, Animation, Comedy, Fantasy]
9738                         [Animation, Comedy, Fantasy]
9739                                              [Drama]
9740                                  [Action, Animation]
9741                                             [Comedy]
Name: genres, Length: 9742, dtype: object

In [5]:
i = 0
values = []
while i < len(genres):
    j = 0
    while j < len(genres[i]):
        if genres[i][j] not in values:
            values.append(genres[i][j])
            j += 1
        else:
            j += 1
    i = i + 1

In [6]:
values # here is all the unique genres in our data

['Adventure',
 'Animation',
 'Children',
 'Comedy',
 'Fantasy',
 'Romance',
 'Drama',
 'Action',
 'Crime',
 'Thriller',
 'Horror',
 'Mystery',
 'Sci-Fi',
 'War',
 'Musical',
 'Documentary',
 'IMAX',
 'Western',
 'Film-Noir',
 '(no genres listed)']

## Next I will look further into our results obtained in Q5 part3

I will be looking at the genres drama and film-noir and try to build on some of our theory from the report.

In [7]:
movies.head()

movieId                               title  \
0        1                    Toy Story (1995)   
1        2                      Jumanji (1995)   
2        3             Grumpier Old Men (1995)   
3        4            Waiting to Exhale (1995)   
4        5  Father of the Bride Part II (1995)   

                                        genres  
0  Adventure|Animation|Children|Comedy|Fantasy  
1                   Adventure|Children|Fantasy  
2                               Comedy|Romance  
3                         Comedy|Drama|Romance  
4                                       Comedy

In [8]:
movies['title'].nunique()

9737

In [11]:
movies['title'].value_counts()

Eros (2004)                                      2
Saturn 3 (1980)                                  2
War of the Worlds (2005)                         2
Emma (1996)                                      2
Confessions of a Dangerous Mind (2002)           2
                                                ..
Revenge of the Nerds IV: Nerds in Love (1994)    1
Wizards (1977)                                   1
Yes Man (2008)                                   1
Small Soldiers (1998)                            1
Ladybird Ladybird (1994)                         1
Name: title, Length: 9737, dtype: int64

In [30]:
def no_movies_genre(input):
    i = 0
    count = 0
    while i < len(genres):
        if input in genres[i]:
            count += 1
            i += 1
        else:
            i += 1
    print('Number of movies containing genre ' + input + ' is: ' + str(count))

In [31]:
no_movies_genre('Film-Noir')

Number of movies containing genre Film-Noir is: 87


In [32]:
no_movies_genre('Drama')

Number of movies containing genre Drama is: 4361


It is interesting to note here that for every film that has the label 'Film-Noir' for genre there is exactly 10 reviewers for each movie (870 movies contained this genre from our pig analysis). While this number is not as high it is still just under 10 for the drama genre. It definitely adds to our theory however that the Film-Noir genre is very unique and the audience it attracts is probably an audience that has a keen interest in films of this type. The drama label however is probably quite vague in the context of so many movies containing this genre type. It could potentially be one of the least relevant labels in a film which is more geared towards comedy or thriller than drama hence the average rating could suffer as a result. We can see from our earlier analysis that only 1053 of the 4361 movies with the genre label are soley dramas. This certainly backs up what we are saying that our drama label can be mixed in amongst other genres and the average ratings can suffer as a result.